File: frontend1/lib/services/user_service.dart
import 'package:frontend1/models/user.dart';
import 'package:frontend1/models/user_stats.dart';
import 'package:frontend1/services/api_service.dart';

class UserService {
  final ApiService _apiService = ApiService();

  Future<List<User>> getUsers() async {
    try {
      final response = await _apiService.get('/auth/users');
      return (response.data as List).map((i) => User.fromJson(i)).toList();
    } catch (e) {
      print('[UserService] Erreur getUsers: $e');
      rethrow;
    }
  }

  Future<UserStats> getUserStats(
    int id,
    String startDate,
    String endDate,
  ) async {
    try {
      final response = await _apiService.get(
        '/users/$id/stats',
        queryParameters: {'start_date': startDate, 'end_date': endDate},
      );
      return UserStats.fromJson(response.data);
    } catch (e) {
      print('[UserService] Erreur getUserStats: $e');
      rethrow;
    }
  }

  Future<User> createUser(Map<String, dynamic> data) async {
    try {
      final response = await _apiService.post(
        '/auth/register',
        data: data,
      ); // Ou /users/create selon backend, React use userService.create qui pointe vers auth/register ou users/
      // React userService.create -> api.post('/users/') usually. Mais si c'est pour register, c'est souvent auth.
      // Backend users.py a POST /users/
      return User.fromJson(response.data);
    } catch (e) {
      print('[UserService] Erreur createUser: $e');
      rethrow;
    }
  }

  Future<User> updateUser(int id, Map<String, dynamic> data) async {
    try {
      final response = await _apiService.put('/users/$id', data: data);
      return User.fromJson(response.data);
    } catch (e) {
      print('[UserService] Erreur updateUser: $e');
      rethrow;
    }
  }

  Future<void> deleteUser(int id) async {
    try {
      await _apiService.delete('/users/$id');
    } catch (e) {
      print('[UserService] Erreur deleteUser: $e');
      rethrow;
    }
  }

  Future<void> toggleStatus(int id) async {
    try {
      await _apiService.post('/users/$id/toggle-status');
    } catch (e) {
      print('[UserService] Erreur toggleStatus: $e');
      rethrow;
    }
  }

  Future<void> updatePassword(int id, String newPassword) async {
    try {
      await _apiService.post(
        '/users/$id/password',
        data: {'password': newPassword},
      );
    } catch (e) {
      print('[UserService] Erreur updatePassword: $e');
      rethrow;
    }
  }
}

File: frontend/src/services/userService.js
import api from './api';

export const userService = {
  /**
   * Get all users (admin only)
   */
  async getAll() {
    const response = await api.get('/auth/users');
    return response.data;
  },

  /**
   * Create a new user
   * @param {Object} userData { username, password, role }
   */
  async create(userData) {
    const response = await api.post('/auth/register', userData);
    return response.data;
  },

  /**
   * Update a user
   * @param {number} userId
   * @param {Object} userData { username, password, role, is_active }
   */
  async update(userId, userData) {
    const response = await api.put(`/users/${userId}`, userData);
    return response.data;
  },

  /**
   * Change user password
   * @param {number} userId
   * @param {string} newPassword
   */
  async updatePassword(userId, newPassword) {
    const response = await api.put(`/auth/users/${userId}/password`, {
      password: newPassword,
    });
    return response.data;
  },

  /**
   * Delete user
   * @param {number} userId
   */
  async delete(userId) {
    const response = await api.delete(`/auth/users/${userId}`);
    return response.data;
  },

  /**
   * Toggle user active status (activate/deactivate)
   * @param {number} userId
   */
  async toggleStatus(userId) {
    const response = await api.put(`/auth/users/${userId}/toggle-status`);
    return response.data;
  },

  /**
   * Get user sales statistics
   * @param {number} userId
   * @param {string} startDate - YYYY-MM-DD format
   * @param {string} endDate - YYYY-MM-DD format
   */
  async getSalesStats(userId, startDate = null, endDate = null) {
    const params = {};
    if (startDate) params.start_date = startDate;
    if (endDate) params.end_date = endDate;
    
    const response = await api.get(`/auth/users/${userId}/sales-stats`, { params });
    return response.data;
  },

  /**
   * Get all users performance comparison
   * @param {string} startDate - YYYY-MM-DD format
   * @param {string} endDate - YYYY-MM-DD format
   */
  async getAllPerformance(startDate = null, endDate = null) {
    const params = {};
    if (startDate) params.start_date = startDate;
    if (endDate) params.end_date = endDate;
    
    const response = await api.get('/auth/users/sales-performance', { params });
    return response.data;
  },
};

export default userService;

